File: lv1_module2_minsikim/ros2_ws/src/turtle_py/turtle_py/distance_subscriber.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import Float32

class DistanceSubscriber(Node):
    def __init__(self):
        super().__init__('distance_subscriber')
        
        self.declare_parameter('warn_distance', 2.5)
        
        self.distance_subscription = self.create_subscription(
            Float32,
            '/turtle_distance',
            self.distance_callback,
            10
        )
        
    def distance_callback(self, msg):
        warn_distance = self.get_parameter('warn_distance').value

        if msg.data > warn_distance:
            self.get_logger().warn(
                f'Distance warning: {msg.data:.3f} m'
            )
        
        
def main(args=None):
    rclpy.init(args=args)

    node = DistanceSubscriber()

    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()

File: lv1_module2_minsikim/ros2_ws/src/turtle_cpp/src/distance_subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/float32.hpp"

class DistanceSubscriber : public rclcpp::Node
{
public:
    DistanceSubscriber()
    : Node("distance_subscriber")
    {
        subscription_ =
            this->create_subscription<std_msgs::msg::Float32>(
                "/turtle_distance",
                10,
                std::bind(
                    &DistanceSubscriber::distance_callback,
                    this,
                    std::placeholders::_1));
    }

private:
    void distance_callback(
        const std_msgs::msg::Float32::SharedPtr msg)
    {
        RCLCPP_INFO(
            this->get_logger(),
            "Distance: %.3f m",
            msg->data);
    }

    rclcpp::Subscription<std_msgs::msg::Float32>::SharedPtr
        subscription_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<DistanceSubscriber>();

    rclcpp::spin(node);

    rclcpp::shutdown();

    return 0;
}